packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -264,6 +416,303 @@ const Dashboard = () => {
+
+For example, we want users to: 
+
+1. Be able to add a “More” dropdown with custom items (It’s just a Menu)
+2. When the screen size reduces, automatically inject extra items into “More” menu
+3. Have the flexibility to customize the what’s inside “More” menu
+
+**Our current API looks like this:**
+
+```jsx
+<TabNav>
+  <TabNavItemLink icon={HomeIcon} accessibilityLabel="Home" href="/home" />
+  <TabNavItemLink href="/payroll">Payroll</TabNavItemLink>
+  <TabNavItemLink href="/payments">Payments</TabNavItemLink>
+  <Menu>
+    <TabNavItemLink href="/more">More</TabNavItemLink>
+    <MenuOverlay>
+      {overflowingItems.map((item) => {
+        // ??
+      })}
+      <MenuItem>
+        <ExploreItem title="Magic Checkout" description="Some product description" />
+      </MenuItem>
+      <MenuItem>
+        <ExploreItem title="RazorX" description="Some product description" />
+      </MenuItem>
+    </MenuOverlay>
+  </Menu>
+</TabNav>
+```
+
+**This API has few issues:**
+
+1. While it lets you add the “More” Menu and choose initial ordering of Magic Checkout & RazorX, when screen size reduces there is no straightforward or intuitive way to move the overflowing items into the “More” Menu
+2. Even if we could somehow magically get the overflowingItems what data will item contain that will enable us to render the <ExploreItem />? We need title, description, icon etc data for rendering which the <TabNavItemLink> may not even need or be provided with since we don’t need to render description or icon. 
+
+----
+
+
+> NOTE: If you want to go into the rabbit hole you can read the alternatives below.
+
+### Alternative APIs
+
+**Table Like Data API (Accepted approach):**
+
+A data driven API where we pass the data for the items and use render props to render the items and the overflowingItems.
+
+**Pros:**
+1. Users have easy access to the overflowingItems via props
+2. Intuitive and faimilar API, similar to Table API
+3. Users have the flexibility to render TabNavItemLinks or any other component.
+4. Users can easily control the order of the items inside the "More" dropdown
+5. We have the data of the items so we can handle the responsiveness internally
+
+**Cons:**
+1. More verbose than compound API
+
+```jsx
+<TabNav
+  items={[
+    { href: '/home', title: 'Home' },
+    { href: '/payroll', title: 'Payroll' },
+    { href: '/payments', title: 'Payments' },
+    {
+      href: '/rize',
+      title: 'rize',
+      description: 'rize description',
+      icon: RizeIcon,
+      // force this item to always be inside "more" regardless of screen size
+      isAlwaysInMore: true,
+    },
+  ]}
+>
+  {({ items, moreItems }) => {
+    return (
+      <>
+        <TabNavItems>
+          {items.map((item) => (
+            <TabNavItem href={item.href}>{item.title}</TabNavItem>
+          ))}
+        </TabNavItems>
+        <Menu>
+          <TabNavItem>More</TabNavItem>
```

Comment:
can this also be in `isActive` state?

Reply:
Also do we have usecase of showing `More: ItemName` here anymore? or will it always stay as More?

Reply:
That that can also have isActive state. And we do have usecase of showing "More: SelectedItemName"

> How does the selected state of item inside More look like?

There's no selection state inside of the more from design side, it's represented via the "More: SelectedItemName", And in future if the OneDashboard team changes the design and decides to have active state of the items inside also they will be able to do it without changing anything on blade since they control the whole Menu.